.github/workflows/doc.yml

```diff
@@ -41,3 +41,32 @@ jobs:
           path: out/doc
       - name: Test
         run: NODE=$(command -v node) make test-doc-ci TEST_CI_ARGS="-p actions --measure-flakiness 9"
+  build-preview:
+    if: github.event.pull_request
+    needs: build-docs
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Check out the gh-pages branch
+        uses: actions/checkout@v3
+        with:
+          # We want to persist credentials so we can push to gh-pages
+          persist-credentials: true
+          ref: gh-pages
+      - name: Erase previous version (if it exists)
+        run: rm -rf "${{ github.event.pull_request.number }}"
+      - name: Download tarball from build job
+        uses: actions/download-artifact@v3
+        with:
+          name: docs
+          path: ${{ github.event.pull_request.number }}
+      - name: Commit the doc preview to gh-pages
+        run: |
+          git config user.email "github-bot@iojs.org"
+          git config user.name "Node.js GitHub Bot"
+          git add "${{ github.event.pull_request.number }}"
+          git commit \
+            -m "Add/Update preview for ${{ github.event.pull_request.html_url }}" \
+            -m "The preview will be available at https://${{ github.repository_owner }}.github.io/${{ github.event.pull_request.base.repo.name }}/${{ github.event.pull_request.number }}/api/"
```

Comment:
I'd also prefer to use either a different repository or a different URL at least that more clearly indicates that these are not official docs, but rather (untrusted) previews.